src/platform/packages/shared/kbn-discover-contextual-components/src/data_types/logs/components/summary_column/utils.tsx

```diff
@@ -15,102 +15,189 @@ import type { SharePluginStart } from '@kbn/share-plugin/public';
 
-const resourceCustomComponentsMap: Partial<
-  Record<keyof ResourceFields, React.ComponentType<FieldBadgeWithActionsProps>>
-> = {
-  [SERVICE_NAME_FIELD]: ServiceNameBadgeWithActions,
-};
-
 export interface ResourceFieldDescriptor {
   ResourceBadge: React.ComponentType<FieldBadgeWithActionsProps>;
   Icon?: () => JSX.Element;
-  name: keyof ResourceFields;
+  name: keyof ResourceFields | keyof TraceFields;
   value: string;
 }
 
+const getResourceBadgeComponent = (
+  name: keyof ResourceFields | keyof TraceFields,
+  core: CoreStart,
+  share?: SharePluginStart
+): React.ComponentType<FieldBadgeWithActionsProps> => {
+  if (name === SERVICE_NAME_FIELD) {
+    return (props: FieldBadgeWithActionsProps) => (
+      <ServiceNameBadgeWithActions {...props} share={share} core={core} />
+    );
+  }
+
+  return FieldBadgeWithActions;
+};
+
+const getResourceBadgeIcon = (
+  name: keyof ResourceFields | keyof TraceFields,
+  fields: Readonly<ResourceFields> | Readonly<TraceFields>
+): (() => React.JSX.Element) | undefined => {
+  switch (name) {
+    case SERVICE_NAME_FIELD:
+      return () => {
+        const { euiTheme } = useEuiTheme();
+        return (
+          <AgentIcon
+            agentName={fields[AGENT_NAME_FIELD] as AgentName}
+            size="m"
+            css={css`
+              margin-right: ${euiTheme.size.xs};
+            `}
+          />
+        );
+      };
+    case EVENT_OUTCOME_FIELD:
+      return () => {
+        const { euiTheme } = useEuiTheme();
+
+        const value = (fields as Readonly<TraceFields>)[name];
+
+        const color = value === 'failure' ? 'danger' : value === 'success' ? 'success' : 'subdued';
+
+        return (
+          <EuiIcon
+            color={color}
+            type="dot"
+            size="s"
+            css={css`
+              margin-right: ${euiTheme.size.xs};
+            `}
+          />
+        );
+      };
+    case TRANSACTION_DURATION_FIELD:
+    case SPAN_DURATION_FIELD:
+      return DurationIcon;
+  }
+};
+
+const getFormattedValue = (
+  name: keyof TraceFields,
+  rawValue: string,
+  dataView: DataView
+): string => {
+  const field = dataView.getFieldByName(name);
+  const formatter = field && dataView.getFormatterForField(field);
+
+  if (formatter) {
+    return formatter.convert(rawValue);
+  } else if (DURATION_FIELDS.includes(name)) {
+    return `${rawValue}µs`;
+  }
+
+  return `${rawValue}`;
+};
+
+export const isTraceDocument = (row: DataTableRecord): row is TraceDocument =>
+  getFieldValue(row, DATASTREAM_TYPE_FIELD) === 'traces';
+
+export const createTraceFields = (
+  row: TraceDocument,
+  dataView: DataView,
+  core: CoreStart,
+  share?: SharePluginStart
+): ResourceFieldDescriptor[] => {
+  const traceFields = getUnformattedFields(row, TRACE_FIELDS);
+  const availableFields = getAvailableTraceFields(traceFields);
+
+  return availableFields.map((name) => ({
+    name,
+    value: getFormattedValue(name, traceFields[name], dataView),
```

Comment:
I think I see why there are issues with filtering and copying now. It looks like `ResourceFieldDescriptor.value` expects an unformatted value which is used for both rendering and actions in `FieldBadgeWithActions`. To properly support rendering formatted values, it probably needs to be split into `rawValue` and `formattedValue` props, and then used accordingly in `FieldBadgeWithActions`.